datastore/common/objects/OperatingSystem.cpp

```diff
@@ -94,8 +100,9 @@ namespace netmeld::datastore::objects {
                 << std::endl;
       return;
     }
-
-    ipAddr.save(t, toolRunId, deviceId);
+    if (ipAddr.isValid()) { 
+      ipAddr.save(t, toolRunId, deviceId);
+    }
```

Comment:
The IP address has to be valid when reaching here, otherwise the logic short circuits prior to ever reaching this point.  Also, the `save()` method already does a validity check prior to doing anything.  So this extra check is, thrice, redundant and not needed here.